src/pages/dashboard/panels/ShokoNews.tsx

```diff
@@ -9,9 +9,21 @@ import { DashboardNewsType } from '@/core/types/api/dashboard';
 
 import { useGetShokoNewsFeedQuery } from '@/core/rtkQuery/externalApi';
 
+const newNewsCheck = (date: string) => {
+  const itemDate = new Date(date);
+  const currentDate = new Date();
+  const oneDayInMilliseconds = 24 * 60 * 60 * 1000;
+  const differenceInMilliseconds = currentDate.getTime() - itemDate.getTime();
+  const differenceInDays = Math.floor(differenceInMilliseconds / oneDayInMilliseconds);
+  return differenceInDays <= 14;
+};
+
 const NewsRow = ({ item }: { item: DashboardNewsType }) => (
   <div className="flex flex-col" key={item.title}>
-    <p className="font-semibold">{item.date_published}</p>
+    <div className="flex gap-x-4">
+      <p className="font-semibold">{item.date_published}</p>
+      <p className="font-semibold text-panel-important">{newNewsCheck(item.date_published) && 'New Post!'}</p>
```

Comment:
This works but the p element is still added to the DOM. Normally you would write
`{newNewsCheck(item.date_published) && (<p className="font-semibold text-panel-important">'New Post!'</p>)}`

Reply:
Yea the checks needs to include the whole element not just the text

Reply:
Arguably, just make a function component inline (in same file) and return either null or p depending on the condition, it will not have the curly braces in JSX which I don't like. For reference like OptionButton here https://github.com/ShokoAnime/Shoko-WebUI/blob/edc1645fb196064ac2d5039adea1bf58df0af123/src/pages/collection/Collection.tsx#L26-L30
It would take a newsitem or just the date and then either return a `<p>` or a null.